core/trino-spi/src/main/java/io/trino/spi/HashUtils.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi;
+
+import static java.lang.Long.rotateLeft;
+
+public class HashUtils
```

Comment:
I don't think we should add this as a library to the SPI.  The code before was part of the abstract base class, but this make it a utility.

Reply:
It was an static method used in multiple unrelated classes throughout the codebase. That is usually considered a bad practice.